src/DIRAC/Core/Security/DiracX.py

```diff
@@ -40,6 +40,8 @@ def addTokenToPEM(pemPath, group):
         token_content = returnValueOrRaise(
             Client(url="Framework/ProxyManager", proxyLocation=pemPath).exchangeProxyForToken()
         )
+        if not token_content:
```

Comment:
Could you just add a comment that this is solely expected when doing the upgrade ? As far as I know, this is the on and only one case where we have a client more recent than the server. Though, should we really for this specific upgrade ? I am starting to wonder whether we are not better off accepting that upgrading a v8 installation from a v9 client is broken. This allows to be stricter with error handling in v9

Reply:
I can certainly add the comment, but note that it's not only for the upgrade: without, no v9 client would be able to work with a v8 server.

Reply:
Hasn't it always been the policy that old clients work with new servers but the reverse isn't true?

Reply:
Sure, and I am not into adding a new requirement here. The change here is anyway so little that I don't see a good point to stop it (it might be useful for other reasons that right now we do not see).

Reply:
It's more that ideally, you *want* to get an error if the token is empty. And besides the specific case of upgrading, to `v9`, I don't see any other reason to return. Given the upgrade to `v9` will be special anyway, I'd be in favor of not doing that at all, and accepting to upgrade with an old client for once

Reply:
OK, if we *want* an error in this case, then this "fix" would really be "wrong". I will close now this PR, and the other connected one for v8,